tendermint/src/amino_types.rs

```diff
@@ -3,27 +3,31 @@
 
 #![allow(missing_docs)]
 
-pub mod block_id;
-pub mod ed25519;
-pub mod message;
-pub mod ping;
-pub mod proposal;
-pub mod remote_error;
-pub mod signature;
-pub mod time;
-pub mod validate;
-pub mod version;
-pub mod vote;
+mod block_id;
+mod ed25519;
+mod message;
+mod ping;
+mod proposal;
+mod remote_error;
+mod signature;
+mod time;
+mod validate;
+mod version;
+mod vote;
```

Comment:
The disadvantage of relying on re-exports of types defined in non-pub modules is rustc will refer to those types by their fully qualified name, which leaves someone trying to debug in the bizarre position that the compiler referring to a path which cannot be accessed and does not exist in rustdoc.

That said, I do sometimes do this myself, but it's usually when I'm trying to create a crate where the code is factored into internal modules but all types are exported at the toplevel (and when I do that, it's generally because the external API surface of the crate is small to begin with).

The other disadvantage is this approach eliminates the ability to have module-level documentation visible in the rustdoc.

Reply:
What would you propose generally? I think `amino_types` module organisation is ripe for a review anyway.

Reply:
Re-export them, but leave the containing modules `pub` if they're part of the public API